cpp/include/raft/common/detail/nvtx.cuh

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2019-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <mutex>
+#include <rmm/cuda_stream_view.hpp>
+#include <string>
+#include <unordered_map>
+
+namespace raft {
+namespace common {
+namespace detail {
+
+/**
+ * @brief An internal struct to store associated state with the color
+ * generator
+ */
+struct ColorGenState {
+  /** collection of all tagged colors generated so far */
+  static std::unordered_map<std::string, uint32_t> allColors;
+  /** mutex for accessing the above map */
+  static std::mutex mapMutex;
+  /** saturation */
+  static constexpr float S = 0.9f;
+  /** value */
+  static constexpr float V = 0.85f;
+  /** golden ratio */
+  static constexpr float Phi = 1.61803f;
+  /** inverse golden ratio */
+  static constexpr float InvPhi = 1.f / Phi;
+};
+
+std::unordered_map<std::string, uint32_t> ColorGenState::allColors;
+std::mutex ColorGenState::mapMutex;
```

Comment:
I've just realized that the implementation copied from cuml's `nvtx.cu` is really meant to be in a source file rather than in the header. What shall we do about it in currently header-only `raft`?